### PR TITLE
Add JSON deserializer utils

### PR DESCRIPTION
**Pull request by matthiasbeyer**
This is a rip from #230, adding some utils for the JSON
deserializer implementation.

### CODE REVIEW

File: src/serialize/json/statemachine.md [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+         \-------------------------------------------------------------/
+
+### Command state nets
+
+This section is splitted into several subsections which contain partial state
+nets each.
+
+#### From initial to Command arguments
+
+The following is a partial state machine of how the command state is reached
```
Comment (neithernut) [inline src/serialize/json/statemachine.md]: `state chart`, not `state machine`, in this case.
And you do far more than just _reaching_ the state.

File: src/serialize/json/statemachine.md [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+
+State diagrams
+--------------
+
+This section contains state diagrams for single paths through the overall state
+net.
+
+### UID state net
+
+The following net describes the states which are passed while parsing a UID
```
Comment (neithernut) [inline src/serialize/json/statemachine.md]: `state chart`, not `net`.

**Comment by neithernut** [comment]
ACK for squashing.

**Comment by matthiasbeyer** [comment]
I wouldn't squash things here. Do you want me to?

**Comment by neithernut** [comment]
Er... yes. For example:
- https://github.com/matthiasbeyer/waysome/commit/4e4a5b219ff354c9e7c30af4aa49ba43faef93d7
- https://github.com/matthiasbeyer/waysome/commit/498c3e1ccf0a94a2058ddae684fe42b7a1e95104

And a few more...

**Comment by neithernut** [comment]
ACK.

**Pull request merged**